packages/atomic/src/components/facets/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -190,11 +193,17 @@ export class AtomicCategoryFacet
     };
     this.facet = buildCategoryFacet(this.bindings.engine, {options});
     this.facetId = this.facet.state.facetId;
-    registerFacetToStore(this.bindings.store, 'categoryFacets', {
-      label: this.label,
-      facetId: this.facetId!,
-      element: this.host,
-    });
+    registerFacetToStore(
+      this.bindings.store,
+      'categoryFacets',
+      {
+        label: this.label,
+        facetId: this.facetId!,
+        facet: this.facet,
+        element: this.host,
+      },
+      parseDependsOn(this.dependsOn)
```

Comment:
A `Record<string, string>` is a pretty generic type, so I decided to parse it here.

I decided not to add dependencies in the object above, because this object directly replaces the facet state in the Atomic store, and I wanted dependencies to be in a separate array.